compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -1184,17 +1184,76 @@ public void visit(BLangLetExpression letExpression) {
 
     @Override
     public void visit(BLangTupleVariable varNode) {
-        result = varNode;
+        //  case 1:
+        //  (string, int) (a, b) = (tuple)
+        //
+        //  any[] x = (tuple);
+        //  string a = x[0];
+        //  int b = x[1];
+        //
+        //  case 2:
+        //  ((string, float) int)) ((a, b), c)) = (tuple)
+        //
+        //  any[] x = (tuple);
+        //  string a = x[0][0];
+        //  float b = x[0][1];
+        //  int c = x[1];
+
+        //create tuple destruct block stmt
+        final BLangBlockStmt blockStmt = ASTBuilderUtil.createBlockStmt(varNode.pos);
+
+        //create a simple var for the array 'any[] x = (tuple)' based on the dimension for x
+        String name = "tuple";
```

Comment:
Not introduced by this PR, but shouldn't we be using different names, starting with `$`?

Reply:
you mean unique names? yeah, have noticed the use of same name in some places in desugar. will change and see if it breaks anything

Reply:
Changed to `$tuple$`